main/modes/mode_jukebox.c

```diff
@@ -0,0 +1,750 @@
+    {.name = "Line 12", .song = &lineTwelveSFX},
+    {.name = "Line 13", .song = &lineThirteenSFX},
+    {.name = "Line 14", .song = &lineFourteenSFX},
+    {.name = "Line 15", .song = &lineFifteenSFX},
+    {.name = "Line 16", .song = &lineSixteenSFX},
+    {.name = "1-Line Clear", .song = &singleLineClearSFX},
+    {.name = "2-Line Clear", .song = &doubleLineClearSFX},
+    {.name = "3-Line Clear", .song = &tripleLineClearSFX},
+    {.name = "4-Line Clear", .song = &quadLineClearSFX},
+    {.name = "Game Over", .song = &gameOverSting},
+};
+
+static const jukeboxSong platformerSfx[] =
+{
+    {.name = "Menu Select", .song = &sndMenuSelect},
+    {.name = "Menu Confirm", .song = &sndMenuConfirm},
+    {.name = "Menu Deny", .song = &sndMenuDeny},
+    {.name = "Start (Unused)", .song = &sndGameStart},
+    {.name = "Jump 1", .song = &sndJump1},
+    {.name = "Jump 2", .song = &sndJump2},
+    {.name = "Jump 3", .song = &sndJump3},
+    {.name = "Hurt", .song = &sndHurt},
+    {.name = "Hit", .song = &sndHit},
+    {.name = "Squish", .song = &sndSquish},
+    {.name = "Break", .song = &sndBreak},
+    {.name = "Coin", .song = &sndCoin},
+    {.name = "Power Up", .song = &sndPowerUp},
+    {.name = "Warp", .song = &sndWarp},
+    {.name = "Wave Ball", .song = &sndWaveBall},
+    {.name = "Death", .song = &sndDie},
+};
+
+static const jukeboxSong jumperSfx[] =
+{
+    {.name = "Jump", .song = &jumpPlayerJump},
+    {.name = "Collect", .song = &jumperPlayerCollect},
+    {.name = "Broke Combo", .song = &jumpPlayerBrokeCombo},
+    {.name = "EvilDonut Jump", .song = &jumpEvilDonutJump},
+    {.name = "Blump Jump", .song = &jumpBlumpJump},
+    {.name = "Countdown", .song = &jumpCountdown},
+};
+
+static const jukeboxSong tunernomesfx[] =
+{
+    {.name = "Primary", .song = &metronome_primary},
+    {.name = "Secondary", .song = &metronome_secondary},
+};
+
+static const jukeboxCategory sfxCategories[] =
+{
+    {.categoryName = "Swadge Bros", .numSongs = 2, .songs = fighterSfx},
+    {.categoryName = "Tiltrads", .numSongs = 22, .songs = tiltradsSfx},
+    {.categoryName = "Swadge Land", .numSongs = 16, .songs = platformerSfx},
+    {.categoryName = "Donut Jump", .numSongs = 6, .songs = jumperSfx},
+    {.categoryName = "Tunernome", .numSongs = 2, .songs = tunernomesfx},
+};
+
+/*============================================================================
+ * Functions
+ *==========================================================================*/
+
+/**
+ * Initializer for jukebox
+ */
+void  jukeboxEnterMode(display_t* disp)
+{
+    // Allocate zero'd memory for the mode
+    jukebox = calloc(1, sizeof(jukebox_t));
+
+    jukebox->disp = disp;
+
+    loadFont("ibm_vga8.font", &jukebox->ibm_vga8);
+    loadFont("radiostars.font", &jukebox->radiostars);
+    loadFont("mm.font", &jukebox->mm);
+
+    loadWsg("arrow12.wsg", &jukebox->arrow);
+
+    jukebox->menu = initMeleeMenu(str_jukebox, &(jukebox->mm), jukeboxMainMenuCb);
```

Comment:
Don't forget to call `deinitMeleeMenu()`!

Reply:
done